### DISCUSSION TITLE
想问下目前对 mysql 的 json 列的支持如何了，是否支持 json 列设置默认值？

### DISCUSSION THREAD

**Discussion by JiChao99** [hhyo/Archery#2554] [Q&A]
demo上的是不支持的，我没具体看是mysql版本，或者验证器用的什么版本；

MySQL 8.0.13 是支持的。

示例脚本

```sql
CREATE TABLE t1 (
  -- literal defaults
  i INT         DEFAULT 0,
  c VARCHAR(10) DEFAULT '',
  -- expression defaults
  f FLOAT       DEFAULT (RAND() * RAND()),
  b BINARY(16)  DEFAULT (UUID_TO_BIN(UUID())),
  d DATE        DEFAULT (CURRENT_DATE + INTERVAL 1 YEAR),
  p POINT       DEFAULT (Point(0,0)),
  j JSON        DEFAULT (JSON_ARRAY())
);
```

ref: https://dev.mysql.com/doc/refman/8.0/en/data-type-defaults.html

**Comment by LeoQuote** [comment]
得看 goinception 的支持情况

**Comment by JiChao99** [comment]
看起来还是不支持的

![6f4c883068c10631fddd8c4338d1b172](https://github.com/hhyo/Archery/assets/15973872/ca6b4049-7097-4d91-9403-1fae676ebcd7)


ref: https://hanchuanchuan.github.io/goInception/rules.html#default-rules
